jinad/api/endpoints/flow.py

```diff
@@ -148,18 +194,29 @@ def destroy_flow(
         flow_wrapper.f.close()
         flow_dict.pop(flow_id)
         return {
-            'status_code': 200
+            'status_code': status.HTTP_200_OK
         }
     except KeyError:
-        raise HTTPException(status_code=404,
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
                             detail=f'Flow ID {flow_id} not found! Please create a new Flow')
 
 
+@router.websocket(
+    path='/wslogs'
+)
+async def _websocket_logs(websocket: WebSocket):
+    # TODO: extend this to work with fluentd logs
+    await websocket.accept()
+    while True:
+        data = await websocket.receive_text()
+        await websocket.send_text(f'Message text: {data}')
+
+
 @router.on_event('shutdown')
-def shutdown():
+def _shutdown():
     global flow_dict
     if flow_dict:
-        for _id, flow in flow_dict:
-            logger.info(f'flow id {_id} still alive! Closing it!')
+        for flow_id, flow in flow_dict.copy().items():
```

Comment:
why is copy needed? just curious

Reply:
This is because I'm popping items in the dictionary inside the loop, hence during iteration the dict changes resulting in error - `RuntimeError: dictionary changed size during iteration`

Reply:
yes good! Thanks!